hazelcast/src/test/java/com/hazelcast/client/impl/spi/impl/ClientClusterServiceImplTest.java

```diff
@@ -0,0 +1,503 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hazelcast.client.impl.spi.impl;
+
+import com.hazelcast.cluster.Address;
+import com.hazelcast.cluster.InitialMembershipEvent;
+import com.hazelcast.cluster.InitialMembershipListener;
+import com.hazelcast.cluster.Member;
+import com.hazelcast.cluster.MembershipEvent;
+import com.hazelcast.cluster.MembershipListener;
+import com.hazelcast.instance.BuildInfoProvider;
+import com.hazelcast.internal.cluster.MemberInfo;
+import com.hazelcast.logging.ILogger;
+import com.hazelcast.test.HazelcastParallelClassRunner;
+import com.hazelcast.test.HazelcastTestSupport;
+import com.hazelcast.test.annotation.ParallelJVMTest;
+import com.hazelcast.test.annotation.QuickTest;
+import com.hazelcast.version.MemberVersion;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import javax.annotation.Nonnull;
+import java.net.UnknownHostException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static java.util.Arrays.asList;
+import static java.util.Collections.emptyMap;
+import static java.util.Collections.singleton;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+
+@RunWith(HazelcastParallelClassRunner.class)
+@Category({QuickTest.class, ParallelJVMTest.class})
+public class ClientClusterServiceImplTest extends HazelcastTestSupport {
+
+    private static final MemberVersion VERSION = MemberVersion.of(BuildInfoProvider.getBuildInfo().getVersion());
+
+    @Test
+    public void testMemberAdded() {
+        LinkedList<Member> members = new LinkedList<>();
+        ClientClusterServiceImpl clusterService = new ClientClusterServiceImpl(mock(ILogger.class));
+        clusterService.addMembershipListener(new MembershipListener() {
+            @Override
+            public void memberAdded(MembershipEvent membershipEvent) {
+                members.add(membershipEvent.getMember());
+            }
+
+            @Override
+            public void memberRemoved(MembershipEvent membershipEvent) {
+
+            }
+        });
+        MemberInfo member = member("127.0.0.1");
+        UUID clusterUuid = UUID.randomUUID();
+        // triggers initial event
+        clusterService.handleMembersViewEvent(1, asList(member), clusterUuid);
+        // triggers member added
+        MemberInfo memberInfo = member("127.0.0.2");
+        clusterService.handleMembersViewEvent(2, asList(member, memberInfo), clusterUuid);
+        assertCollection(members, Collections.singleton(memberInfo.toMember()));
```

Comment:
Minor suggestion, we can change the order of assert statements, since that would be more "granular to specific" asserts.

Reply:
Since this is a backport a prefer not to do a change if it is not something we can't live with. This concern seems minor as you also stated.